### PR TITLE
YLP-380 Fix peer review

### PR DESCRIPTION
**Pull request by yvon-dblg**

### CODE REVIEW

File: packages/yourloops/pages/hcp/team-edit-dialog.tsx [hunk 1]
```diff
@@ -195,129 +215,157 @@ function TeamEditModal(props: TeamEditModalProps): JSX.Element {
...
+
+    setAddrCity(team.address?.city ?? "");
+    setAddrCountry(team.address?.country ?? auth.user?.settings?.country ?? "FR");
+    setAddrLine1(team.address?.line1 ?? "");
+    setAddrLine2(team.address?.line2 ?? "");
+    setAddrZipCode(team.address?.zip ?? "");
+    setTeamEmail(team.email ?? "");
+    setTeamName(team.name ?? "");
+    setTeamPhone(team.phone ?? "");
+  }
```
Comment (leovalette) [inline packages/yourloops/pages/hcp/team-edit-dialog.tsx:235]: Should be in a useEffect

File: packages/yourloops/pages/hcp/team-edit-dialog.tsx [hunk 2]
```diff
@@ -136,42 +145,53 @@ function TeamEditModal(props: TeamEditModalProps): JSX.Element {
...
     };
-    if (addrLine2.length < 1) {
-      delete team.address?.line2;
+    if (line2.length < 1) {
+      delete updatedTeam.address.line2;
     }
-    onSaveTeam(team);
+    onSaveTeam(updatedTeam);
   };
   const handleChangeTeamName = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>): void => {
```
Comment (leovalette) [inline packages/yourloops/pages/hcp/team-edit-dialog.tsx:196]:
```suggestion
  
  const handleChange = (
    setState: React.Dispatch<React.SetStateAction<string>>
  ): ((event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => void) => (event) => setState(event.target.value);
```

Comment (leovalette) [reply packages/yourloops/pages/hcp/team-edit-dialog.tsx:196]: I wanted to select lines 196 to 216

File: packages/yourloops/pages/hcp/team-edit-dialog.tsx [hunk 3]
```diff
@@ -136,42 +145,53 @@ function TeamEditModal(props: TeamEditModalProps): JSX.Element {
...
+      const len = value.length;
+      return len > limits.min && len < limits.max;
+    };
+    let valid = inLimit(teamName.trim(), teamFieldsLimits.name);
+    valid = valid && inLimit(teamPhone.trim(), teamFieldsLimits.phone);
+    valid = valid && inLimit(addrLine1.trim(), teamFieldsLimits.addLine1);
+    valid = valid && inLimit(addrLine2.trim(), teamFieldsLimits.addLine2);
+    valid = valid && inLimit(addrZipCode.trim(), teamFieldsLimits.zipCode);
+    valid = valid && inLimit(addrCity.trim(), teamFieldsLimits.city);
+    valid = valid && inLimit(addrCountry.trim(), teamFieldsLimits.country);
```
Comment (leovalette) [inline packages/yourloops/pages/hcp/team-edit-dialog.tsx:158]: This looks weird to me...

**Review by leovalette — CHANGES_REQUESTED** [review]

File: packages/yourloops/pages/hcp/team-card.tsx [hunk 4]
```diff
@@ -107,46 +116,59 @@ const teamInfoStyles = makeStyles((theme: Theme) => {
...
+  );
+}
+
 function TeamInfo(props: TeamInfoProps): JSX.Element | null {
   const { label, value, icon } = props;
   const classes = teamInfoStyles();
   const { t } = useTranslation("yourloops");
 
-  if (typeof value !== "string") {
+  if (_.isUndefined(value)) {
```
[the remark this answers is not in the document]
Comment (yvon-dblg) [reply packages/yourloops/pages/hcp/team-card.tsx:142]: I don't like to === undefined.
According to the doc I've read, it's not safe.

File: packages/yourloops/pages/hcp/team-leave-dialog.tsx [hunk 5]
```diff
@@ -141,37 +130,45 @@ function LeaveTeamDialogContent(props: LeaveTeamDialogElementsProps): JSX.Elemen
...
+  } else if (userIsTheOnlyAdministrator) {
+    consequences = (
+      <DialogContentText id="team-leave-dialog-consequences">
+        <Trans
+          i18nKey="team-leave-dialog-only-admin-consequences"
+          t={t}
+          components={{ strong: <strong /> }}
+          parent={React.Fragment}>
+          Since you are the only team&apos;s administrator, you need to
+          <strong>give administrator permission to another member</strong> before leaving.
```
[the remark this answers is not in the document]
Comment (yvon-dblg) [reply packages/yourloops/pages/hcp/team-leave-dialog.tsx:157]: Nope, see the tag property: i18nKey

File: packages/yourloops/pages/hcp/team-edit-dialog.tsx (hunk 3 above)
Comment (yvon-dblg) [reply packages/yourloops/pages/hcp/team-edit-dialog.tsx:158]: what is weird ?

File: packages/yourloops/pages/hcp/team-edit-dialog.tsx (hunk 2 above)
Comment (yvon-dblg) [reply packages/yourloops/pages/hcp/team-edit-dialog.tsx:196]: Sorry I don't understand this complex line.
Having a function, which return a function (x2) make things hard to understand.

File: packages/yourloops/pages/hcp/team-card.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (leovalette) [reply packages/yourloops/pages/hcp/team-card.tsx:142]: Lodash isUndefined method is a wrapper of `=== undefined` : [isUndefined](https://github.com/lodash/lodash/blob/master/isUndefined.js)

Where can i find the doc you read about this ? (just being curious)

File: packages/yourloops/pages/hcp/team-leave-dialog.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (leovalette) [reply packages/yourloops/pages/hcp/team-leave-dialog.tsx:157]: Ok I need to take a deeper look at the Trans component

File: packages/yourloops/pages/hcp/team-card.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (yvon-dblg) [reply packages/yourloops/pages/hcp/team-card.tsx:142]: For the reference, I need to search again. I've read this 2 years ago...

**Review by leovalette — APPROVED** [review]
LGTM, some parts of the code maybe should be moved in useMemo/useEffect blocks. But it can be done later

File: packages/yourloops/pages/hcp/team-members-table.tsx [hunk 6]
```diff
@@ -69,23 +69,21 @@ const teamMembersStyles = makeStyles((theme: Theme) => {
   };
 });
 
-function PersonRemoveIcon(): JSX.Element {
+function PersonRemoveIcon(props: SvgIconProps): JSX.Element {
```
Comment (franck-fourel) [inline packages/yourloops/pages/hcp/team-members-table.tsx:72]: same comment as `VerifiedIcon`in team-card

**Review by franck-fourel — COMMENTED** [review]
couple of comments that can be discussed

File: packages/yourloops/pages/hcp/team-members-table.tsx (hunk 6 above)
Comment (yvon-dblg) [reply packages/yourloops/pages/hcp/team-members-table.tsx:72]: Will we fix in another PR

File: packages/yourloops/pages/hcp/team-card.tsx [hunk 7]
```diff
@@ -107,23 +116,36 @@ const teamInfoStyles = makeStyles((theme: Theme) => {
...
+  // Source: https://material.io/resources/icons/?icon=verified&style=baseline
+  // prettier-ignore
+  return (
+    <SvgIcon xmlns="http://www.w3.org/2000/svg" enableBackground="new 0 0 24 24" viewBox="0 0 24 24" width="24px" height="24px" {...props}>
+      <g><rect fill="none" height="24" width="24"/></g>
+      <g><path d="M23,12l-2.44-2.79l0.34-3.69l-3.61-0.82L15.4,1.5L12,2.96L8.6,1.5L6.71,4.69L3.1,5.5L3.44,9.2L1,12l2.44,2.79l-0.34,3.7 l3.61,0.82L8.6,22.5l3.4-1.47l3.4,1.46l1.89-3.19l3.61-0.82l-0.34-3.69L23,12z M10.09,16.72l-3.8-3.81l1.48-1.48l2.32,2.33 l5.85-5.87l1.48,1.48L10.09,16.72z"/></g>
+    </SvgIcon>
+  );
+}
+
```
[the remark this answers is not in the document]
Comment (yvon-dblg) [reply packages/yourloops/pages/hcp/team-card.tsx:136]: Will we fix in another PR

File: packages/yourloops/pages/hcp/team-leave-dialog.tsx [hunk 8]
```diff
@@ -85,6 +84,7 @@ function isUserIsTheOnlyAdministrator(team: Team | null, userId: string): boolea
   const members = team.members ?? [];
   let nAdmin = 0;
   let userIsAdmin = false;
+
```
[the remark this answers is not in the document]
Comment (yvon-dblg) [reply packages/yourloops/pages/hcp/team-leave-dialog.tsx:87]: Will be fix in another PR: Team management utils functions will be moved in a dedicated package.

File: packages/yourloops/pages/hcp/team-edit-dialog.tsx [hunk 9]
```diff
@@ -136,188 +144,207 @@ function TeamEditModal(props: TeamEditModalProps): JSX.Element {
...
-    team.address = {
-      line1: addrLine1,
-      line2: addrLine2,
-      zip: addrZipCode,
-      city: addrCity,
-      country: addrCountry,
+    const line2 = addrLine2.trim();
+    updatedTeam.address = {
+      line1: addrLine1.trim(),
+      line2,
```
[the remark this answers is not in the document]
Comment (yvon-dblg) [reply packages/yourloops/pages/hcp/team-edit-dialog.tsx:197]: Ha... yes
I should know that's in Javascript we do not want to do smalls optimizations.
I will change it

File: packages/yourloops/pages/hcp/team-edit-dialog.tsx [hunk 10]
```diff
@@ -95,27 +91,39 @@ const modalStyles = makeStyles((theme: Theme) => {
...
+const teamFieldsLimits = {
+  name: { min: 1, max: 64 },
+  phone: { min: 3, max: 32 },
+  addLine1: { min: 1, max: 128 },
+  addLine2: { min: -1, max: 128 },
+  zipCode: { min: 1, max: 16 },
+  city: { min: 1, max: 128 },
+  country: { min: 1, max: 4 },
+  email: { min: 0, max: 64 },
+};
```
[the remark this answers is not in the document]
Comment (yvon-dblg) [reply packages/yourloops/pages/hcp/team-edit-dialog.tsx:103]: I like the idea, I don't like the idea to add one more deps.
We will think over it after ?

File: packages/yourloops/models/team.ts [hunk 11]
```diff
@@ -20,8 +20,16 @@
 
 import { User } from "./shoreline";
 
-export type TeamType = "medical" | "personal";
-export type TeamMemberRole = "admin" | "viewer" | "patient";
+export enum TeamType {
+  medical = "medical",
+  personal = "personal",
```
[the remark this answers is not in the document]
Comment (yvon-dblg) [reply packages/yourloops/models/team.ts:25]: In rework in next PR

**Review by franck-fourel — APPROVED** [review]
LGTM

**Pull request merged**